test/unit/carousel.ts

```diff
@@ -0,0 +1,414 @@
+
+        expect(removeEventListener).to.be.calledWithExactly('resize', carousel.update);
+      });
+    });
+
+    describe('moveNext', () => {
+      it('should call slideHandler function with correct slide number', () => {
+        carousel.state.currentSlide = 0;
+        carousel.slideHandler = spy();
+
+        carousel.moveNext();
+
+        expect(carousel.slideHandler).to.be.calledWithExactly(1);
+      });
+    });
+
+    describe('movePrevious', () => {
+      it('should call slideHandler function with correct slide number', () => {
+        carousel.state.currentSlide = 0;
+        carousel.slideHandler = spy();
+
+        carousel.movePrevious();
+
+        expect(carousel.slideHandler).to.be.calledWithExactly(-1);
+      });
+    });
+
+    describe('isSwipeToNext()', () => {
+      it('should move next when swiping left with an angle between 315 and 360', () => {
+        const direction = Carousel.isSwipeToNext({ startX: 0, startY: 0 }, -25, 4);
+        expect(direction).to.be.true;
+      });
+
+      it('should move next when swiping left with an angle between 0 and 45', () => {
+        const direction = Carousel.isSwipeToNext({ startX: 0, startY: 0 }, -25, -10);
+        expect(direction).to.be.true;
+      });
+
+      it('should move previous when swiping right with an angle between 135 and 225', () => {
+        const direction = Carousel.isSwipeToNext({ startX: 0, startY: 0 }, 25, 10);
+        expect(direction).to.be.false;
+      });
+
+      it('should not return anything if angle is between 45 to 135 or between225 and 315', () => {
+        const direction = Carousel.isSwipeToNext({ startX: 0, startY: 0 }, 0, 100);
+        expect(direction).to.be.null;
+      });
+    });
+
+    describe('onTouchStart()', () => {
+      it('should add listener', () => {
+        const event = {
+          touches: [{ pageX: 0, pageY: 5 }],
+          target: {
+            addEventListener: spy(),
+            removeEventListener: spy()
+          }
+        };
+        carousel.set = () => {
+          carousel.state.touchObject = result;
+        };
+        const result = { startX: 0, startY: 5 };
+
+        expect(event.target.addEventListener).to.not.be.called;
+        carousel.onTouchStart(event as any);
+        expect(carousel.state.touchObject).to.eql(result);
+        expect(event.target.addEventListener).to.be.calledWithExactly(
+          'touchend',
+          carousel.onTouchEnd
+        );
+      });
+    });
+
+    describe('onTouchEnd()', () => {
+      beforeEach(() => carousel.state.touchObject = { startX: 0, startY: 5 });
+
+      it('should swipe to next slides if swipe distance is long enough', () => {
+        const event = {
+          changedTouches: [{ pageX: 30, pageY: 15 }],
+          target: {
+            addEventListener: spy(),
+            removeEventListener: spy()
+          }
+        };
+        stub(Carousel, 'isSwipeToNext').returns(true);
+        carousel.moveNext = spy();
```

Comment:
doesn't appear as though `moveNext()` or `movePrevious()` are ever called, do we need to make them `spy()`s